### PR TITLE
Make playground work with apps behind a proxy with root_path

### PR DESCRIPTION
**Pull request by nat-n**
Make the playground respect the [root_path](https://fastapi.tiangolo.com/advanced/behind-a-proxy/) from the request scope. 

Also streamline some related code in APIHandler around normalizing trailing slashes on paths.

### CODE REVIEW

**Comment by jlev** [comment]
I was *just* struggling with this with our dev deployment, and was about to open an issue. Love using bleeding edge software!

**Comment by jlev** [comment]
FWIW, this change wasn't sufficient to work for me. I'm starting FastAPI with uvicorn and setting the root-path on the command line, so that setting isn't accessible to the app itself. It IS in the `request.scope["root_path"]` coming from starlette, so I needed to prepend that to the self._base_url on line 1335 of api_handler.py

Not sure if this is the "right way" to do it :shrug:

**Comment by nat-n** [comment]
> FWIW, this change wasn't sufficient to work for me. I'm starting FastAPI with uvicorn and setting the root-path on the command line, so that setting isn't accessible to the app itself. It IS in the `request.scope["root_path"]` coming from starlette, so I needed to prepend that to the self._base_url on line 1335 of api_handler.py ([jlev@5fe0b33](https://github.com/jlev/langserve/commit/5fe0b336ffe0394dc574d7e423e53dbde318d616))
> 
> Not sure if this is the "right way" to do it 🤷

I see. I assumed it was preferable to configure the `root_path` explicitly in the app, such as via an environment variable. But now that you mention it, using the request scope might be better. I'll play around with it and update this PR based on what I learn.

**Comment by eyurtsev** [comment]
@nat-n I added you to the clat-bot list. feel free to ping me when you have a solution in mind and thank you!

**Comment by nat-n** [comment]
@eyurtsev Ready for review. 

Thanks @jlev for the feedback, I usually prefer working with explicit configuration, but in the end I couldn't find any downside to using the root_path from the scope :)

File: tests/unit_tests/test_server_client.py [hunk 1]
```diff
@@ -307,6 +307,42 @@ async def test_serve_playground_with_api_router() -> None:
     assert response.status_code == 200
 
 
+async def test_root_path_on_playground(event_loop: AbstractEventLoop) -> None:
+    """Test that the playground respects the root_path for requesting assets"""
+
+    for root_path in ("/home/root", "/home/root/"):
```
Comment (eyurtsev) [inline tests/unit_tests/test_server_client.py:313]: What's the purpose of looping twice through the same root path?

Comment (nat-n) [reply tests/unit_tests/test_server_client.py:313]: I introduced some logic around slash stripping/joining, so though it was worth ensuring it continues to work with and without the trailing slash.

Comment (eyurtsev) [reply tests/unit_tests/test_server_client.py:313]: ah got it -- i didn't notice the extra `/`

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**